src/IB/IBFESurfaceMethod.cpp

```diff
@@ -1107,22 +1812,40 @@ IBFESurfaceMethod::spreadForce(const int f_data_idx,
                                const double data_time)
 {
     TBOX_ASSERT(MathUtilities<double>::equalEps(data_time, d_half_time));
+    batch_vec_ghost_update({ d_F_half_vecs, d_X_half_vecs, d_P_jump_half_vecs }, INSERT_VALUES, SCATTER_FORWARD);
+
     for (unsigned int part = 0; part < d_num_parts; ++part)
     {
         PetscVector<double>* X_vec = d_X_half_vecs[part];
         PetscVector<double>* X_ghost_vec = d_X_IB_ghost_vecs[part];
         PetscVector<double>* F_vec = d_F_half_vecs[part];
         PetscVector<double>* F_ghost_vec = d_F_IB_ghost_vecs[part];
-        X_vec->localize(*X_ghost_vec);
-        F_vec->localize(*F_ghost_vec);
+        copy_and_synch(*X_vec, *X_ghost_vec);
+        copy_and_synch(*F_vec, *F_ghost_vec);
```

Comment:
We should move this communication out of the loop over different parts. In `IBFEMethod::interpolateVelocity` we do
```cpp

    batch_vec_copy(X_vecs, d_X_IB_ghost_vecs);
    batch_vec_ghost_update(d_X_IB_ghost_vecs, INSERT_VALUES, SCATTER_FORWARD);
```
which will have the same effect but work much better with multiple parts.